memeolist/app/src/main/java/org/aerogear/android/app/memeolist/ui/MemeListActivity.java

```diff
@@ -54,49 +51,75 @@ protected void onCreate(Bundle savedInstanceState) {
 
         ButterKnife.bind(this);
 
-        apolloClient = SyncService.getInstance().getApolloClient();
-
         mMemes.setLayoutManager(new LinearLayoutManager(this));
         new LastAdapter(memes, BR.meme)
                 .map(Meme.class, R.layout.item_meme)
                 .into(mMemes);
 
-        mSwipe.setOnRefreshListener(() -> retrieveMemes());
+        mSwipe.setOnRefreshListener(this::retrieveMemes);
+
+        subscribeMemes();
+        retrieveMemes();
     }
 
-    @Override
-    protected void onStart() {
-        super.onStart();
+    private void subscribeMemes() {
+
+        SyncService.getInstance().subscribe(new NewMemeCreatedSubscription())
```

Comment:
[Passos] Do you think we should use this by default? It will be nice to have this API as addition to already available apollo api, however it will be easier to rely on existing non reactive api that comes from apollo in the memolist app.

Reply:
I'd say, let's document both but let developer free to choice use those wrappers or the Apollo client direct. I'd like to have it here because the Apollo client already has an example on its repo.